c7n/resources/sns.py

```diff
@@ -65,8 +88,69 @@ class SNSCrossAccount(CrossAccountAccessFilter):
                       - permitted-account-01
                       - permitted-account-02
     """
+
+    valid_protocols = (
+        "http",
+        "https",
+        "email",
+        "email-json",
+        "sms",
+        "sqs",
+        "application",
+        "lambda"
+    )
+
+    schema = type_schema(
+        'cross-account',
+        rinherit=CrossAccountAccessFilter.schema,
+        whitelist_endpoints={'type': 'array', 'items': {'type': 'string'}},
+        whitelist_endpoints_from=ValuesFrom.schema,
+        whitelist_protocols={'type': 'array', 'items': {'type': 'string', 'enum': valid_protocols}},
+        whitelist_protocols_from=ValuesFrom.schema
+    )
+
     permissions = ('sns:GetTopicAttributes',)
 
+    checker_factory = SNSPolicyChecker
+
+    def process(self, resources, event=None):
+        self.everyone_only = self.data.get('everyone_only', False)
```

Comment:
for easing subclass responsibility, it might be nicer to move the bulk of this implementation to a get_checker_config and get_checker methods, that a subclass can use super on instead of duplicating to get the base class logic.